dlt/sources/sql_database/arrow_helpers.py

```diff
@@ -1,15 +1,14 @@
 from typing import Any, Sequence
 
 from dlt.common.schema.typing import TTableSchemaColumns
-
 from dlt.common.configuration import with_config
+from dlt.common.configuration.container import Container
 from dlt.common.destination import DestinationCapabilitiesContext
 
 
 @with_config
 def row_tuples_to_arrow(
     rows: Sequence[Any],
-    caps: DestinationCapabilitiesContext = None,
```

Comment:
`with_config` decorator will inspect the signature of `row_tuples_to_arrow` and will inject `caps` automatically (caps derive from `ContainerInjectableContext` which request injection from `Container` which is also a config provider in `dlt`.)

to check that try wrap your test with
```py
with Container().injectable_context(destination_caps):
...
``
and you'll see `destination_caps` injected. (just explaining, see below)

btw. you code below is OK and we can keep it. just remove `@with_config` which is not needed and will slightly slower all calls to decorated function.